### PR TITLE
Add `SpectralModel.spectral_index_error`

### PR DESCRIPTION
**Pull request by AtreyeeS**
Signed-off-by: Atreyee Sinha <asinha@ucm.es>

As brought up by @chaimain on the slack help channel today, error propagation on the index at the particular energy was not implement earlier...

### CODE REVIEW

File: gammapy/modeling/models/spectral.py [hunk 1]
```diff
@@ -536,6 +536,25 @@ def spectral_index(self, energy, epsilon=1e-5):
...
 
+    def spectral_index_error(self, energy, epsilon=1e-5):
+        """Evaluate the error on spectral index at the given energy
+
+        Parameters
+        ----------
+        energy : `~astropy.units.Quantity`
+            Energy at which to estimate the index
+        epsilon : float
+            Fractional energy increment to use for determining the spectral index.
```
Comment (registerrier) [inline gammapy/modeling/models/spectral.py:547]: specify default value.

**Review by registerrier — APPROVED** [review]
Thanks @AtreyeeS . This looks good.

Is there a possibility to have a test that guarantees that the tested value is the expected one? Or is this specific set of parameters and covariance matrix validated from another tool?

Maybe rebase the PR on main to avoid the black CI fails.

**Review by QRemy — APPROVED** [review]
Thanks @AtreyeeS, looks good.

**Review by registerrier — APPROVED** [review]
Thanks @AtreyeeS . No further comments.

**Pull request merged**